sql/event_log.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt@cockroachlabs.com)
+
+package sql
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/roachpb"
+	"github.com/cockroachdb/cockroach/sql/privilege"
+)
+
+// EventLogType represents an event type that can be recorded in the event log.
+type EventLogType string
+
+const (
+	// EventLogCreateTable is recorded when a table is created.
+	EventLogCreateTable EventLogType = "create_table"
+)
+
+// eventTableSchema describes the schema of the event log table.
+const eventTableSchema = `
+CREATE TABLE system.eventlog (
+  timestamp    TIMESTAMP  NOT NULL,
+  eventType    STRING     NOT NULL,
+  effectingID  INT        NOT NULL,
+  reportingID  INT        NOT NULL,
+  info         STRING,
+  uniqueID     INT        DEFAULT experimental_unique_int(),
+  PRIMARY KEY (timestamp, uniqueID)
```

Comment:
It would be nice to be able to query by object name and verb - to have them broken out as separate columns. For example, the name of a table as the object and the "CREATE TABLE" as the verb.